src/runtime_src/tools/xclbinutil/XclBinSignature.cxx

```diff
@@ -482,6 +484,9 @@ void verifyXclBinImage(const std::string& _fileOnDisk,
     throw std::runtime_error("ERROR: Can't add certificate.");
   }
 
+  // -- v3 Certificates with "extendedKeyUsage = codeSigning,1.3.6.1.4.1.2312.16.1.2"
+  // -- e.g. UEFI DKMS module signing certiciate /var/lib/shim-signed/mok/MOK.der
+  X509_STORE_set_purpose(store, XKU_CODE_SIGN);
```

Comment:
This seems to be a fundamental missing step.  Would you be so kind as to point me to where you got this information.